Extensions/BBText/bbtextruntimeobject.js

```diff
@@ -0,0 +1,169 @@
+) {
+  if (initialInstanceData.customSize) {
+    this.setWrappingWidth(initialInstanceData.width);
+  }
+};
+
+gdjs.BBTextRuntimeObject.prototype.onDestroyFromScene = function(runtimeScene) {
+  gdjs.RuntimeObject.prototype.onDestroyFromScene.call(this, runtimeScene);
+
+  this._renderer.onDestroy();
+};
+
+gdjs.BBTextRuntimeObject.prototype.update = function(runtimeScene) {
+  this._renderer.ensureUpToDate();
+};
+
+/**
+ * Set BBText base style properties
+ */
+gdjs.BBTextRuntimeObject.prototype.setBBText = function(text) {
+  this._renderer.setBBText(text);
+};
+
+gdjs.BBTextRuntimeObject.prototype.getBBText = function(text) {
+  console.log(this._renderer);
+  return this._renderer.text;
+};
+
+gdjs.BBTextRuntimeObject.prototype.setColor = function(value) {
+  this._renderer.setBaseProperty('color', value);
+};
+
+gdjs.BBTextRuntimeObject.prototype.setOpacity = function(value) {
+  this._renderer.setBaseProperty('opacity', value);
+};
+
+gdjs.BBTextRuntimeObject.prototype.getOpacity = function() {
+  console.log(this);
+  return this._opacity;
+};
+
+gdjs.BBTextRuntimeObject.prototype.setFontSize = function(value) {
+  this._renderer.setBaseProperty('font size', value);
+};
+
+gdjs.BBTextRuntimeObject.prototype.setFontFamily = function(value) {
+  this._renderer.setBaseProperty('font family', value);
+};
+
+gdjs.BBTextRuntimeObject.prototype.setAlignment = function(value) {
+  this._renderer.setBaseProperty('alignment', value);
+};
+/**
+ * Set object position on X axis.
+ * @param {number} x The new position X of the object.
+ */
+gdjs.BBTextRuntimeObject.prototype.setX = function(x) {
+  gdjs.RuntimeObject.prototype.setX.call(this, x);
+  this._renderer.updatePosition();
+};
+
+/**
+ * Set object position on Y axis.
+ * @param {number} y The new position Y of the object.
+ */
+gdjs.BBTextRuntimeObject.prototype.setY = function(y) {
+  gdjs.RuntimeObject.prototype.setY.call(this, y);
+  this._renderer.updatePosition();
+};
+
+/**
+ * Set the angle of the object.
+ * @param {number} angle The new angle of the object.
+ */
+gdjs.BBTextRuntimeObject.prototype.setAngle = function(angle) {
+  gdjs.RuntimeObject.prototype.setAngle.call(this, angle);
+  this._renderer.updateAngle();
+};
+
+/**
+ * Set object opacity.
+ * @param {number} opacity The new opacity of the object (0-255).
+ */
+gdjs.BBTextRuntimeObject.prototype.setOpacity = function(opacity) {
+  this._opacity = opacity;
+  this._renderer.updateOpacity();
+};
+
+/**
+ * Get object opacity.
+ */
+gdjs.BBTextRuntimeObject.prototype.getOpacity = function() {
+  return this._opacity;
+};
+
+/**
+ * Set the word wrapping width for the text object.
+ * @param {number} width The new width to set.
+ */
+gdjs.BBTextRuntimeObject.prototype.setWrappingWidth = function(width) {
+  if (width <= 1) width = 1;
+  this._wrappingWidth = width;
+};
+
+/**
+ * Set the width.
+ * @param {number} width The new width in pixels.
+ */
+gdjs.BBTextRuntimeObject.prototype.setWidth = function(width) {
```

Comment:
Not sure if this method is needed, I would only keep "setWrappingWidth" like in https://github.com/4ian/GDevelop/blob/master/Extensions/TextObject/textruntimeobject.js